### DISCUSSION TITLE
Is this comment correct?

### DISCUSSION THREAD

**Discussion by bartekw2213** [google/googletest#4521] [Community Help]
My question regards `::testing::internal::EqHelper::Compare` function and the comment added to it.

```
  template <
      typename T1, typename T2,
      // Disable this overload for cases where one argument is a pointer
      // and the other is the null pointer constant.
      typename std::enable_if<!std::is_integral<T1>::value ||
                              !std::is_pointer<T2>::value>::type* = nullptr>
  static AssertionResult Compare(const char* lhs_expression,
                                 const char* rhs_expression, const T1& lhs,
                                 const T2& rhs) {
    return CmpHelperEQ(lhs_expression, rhs_expression, lhs, rhs);
  }
```

In `std::enable_if` we're clearly checking if the first type isn't equal to integral type or second type isn't equal to pointer type.

However, the comment above says that we're disabling an overload when one argument is a pointer and the other is the null pointer constant - so:

- why we're checking if the first type is an integral type?
- why we don't have any check connected with null pointer constant?

**Comment by Shriharsh-Deshmukh** [comment]
The `std::enable_if` check in `::testing::internal::EqHelper::Compare` is designed to prevent comparisons between integral types (like `int`) and pointer types, which could lead to ambiguity, particularly when comparing `0` (a common null pointer constant) with pointers.

1. **Why Check `std::is_integral<T1>`?**: This prevents comparisons between an integer (like `0`) and a pointer, which is prone to errors.

2. **Why No Explicit Null Pointer Check?**: The check is broader, catching all integral-to-pointer comparisons, not just null pointer constants, ensuring safe comparisons across the board.

**Comment by Hasan0971** [comment]
Yes
